pac4j-jwt/src/main/java/org/pac4j/jwt/profile/JwtGenerator.java

```diff
@@ -67,22 +67,34 @@ public JwtGenerator(final String signingSecret, final String encryptionSecret) {
         this.signingSecret = signingSecret;
         this.encryptionSecret = encryptionSecret;
     }
-
+    
     /**
      * Generates a JWT from a user profile.
      *
      * @param profile the given user profile
      * @return the created JWT
      */
     public String generate(final U profile) {
+    	return generate(profile, null, this.jwsAlgorithm);
```

Comment:
It says
+ if (signer == null) {
 +        	// Create HMAC signer
 +        	signer = new MACSigner(this.signingSecret);
 +        }

So I don't see how it's a backward compatibility issue